### PR TITLE
Hot loading asset folder on development

### LINKED ISSUE #540: Assets directory instant refresh
Currently updates to the assets directory require a new `pc run` to copy the files to the `.web/_static` directory. We should have this done automatically (we used symlinks before but they didn't work well on Windows which is why we switched to this way).

### PR DESCRIPTION
**Pull request by PeterYusuke**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [x] Have you linted your code locally prior to submission?

### Description

Add feature for watching `assets` folder changes, and do same changes on `.web/public` folder.
To work this just execute `pc run` with development, but not production.

Close #540.

### CODE REVIEW

File: pynecone/utils.py [hunk 1]
```diff
@@ -622,6 +623,10 @@ def run_frontend(app: App, root: Path, port: str):
     # Set up the frontend.
     setup_frontend(root)
 
+    # start watching asset folder
+    asset_watch = AssetFolderWatch(root)
+    asset_watch.start()
```
Comment (picklelo) [inline pynecone/utils.py:628]: Do we need to stop the watch? For example when they ctrl+c, will this still be watching?

File: pynecone/watch.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    def start(self):
+        """Start watching asset folder."""
+        self.observer = Observer()
+        self.observer.schedule(self.event_handler, self.path, recursive=True)
+        self.observer.start()
+
+
+class AssetFolderHandler(FileSystemEventHandler):
+    """Asset folder event handelr."""
```
Comment (picklelo) [inline pynecone/watch.py:34]: Typo: handler

File: pynecone/watch.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        """Initialize the AssetFolderHandler Class.
+
+        Args:
+            root: root path of the public.
+        """
+        super().__init__()
+        self.root = root
+
+    def on_modified(self, event: FileSystemEvent):
+        """Event hander when a file or folder was modified.
```
Comment (picklelo) [inline pynecone/watch.py:46]: handler

File: pynecone/watch.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        This is called every time after a file is created, modified and deleted.
+
+        Args:
+            event: Event infomation.
```
Comment (picklelo) [inline pynecone/watch.py:50]: typo: information

File: pyproject.toml [hunk 5]
```diff
@@ -49,6 +49,7 @@ toml = "^0.10.2"
 pytest-asyncio = "^0.20.1"
 black = "^22.10.0"
 ruff = "^0.0.244"
+watchdog = "^2.3.1"
```
Comment (picklelo) [inline pyproject.toml:52]: Will we need this in the main dependencies since we need it to be installed in `pip install pynecone`

**Review by picklelo — COMMENTED** [review]
Awesome work, we really needed this!

Just have 2 main questions and some minor typo comments.

File: pynecone/utils.py (hunk 1 above)
Comment (PeterYusuke) [reply pynecone/utils.py:628]: We don't need to do that. 
This is a daemon thread so will be killed when the main thread is exited even if it is caused with ctrl+c.

**Comment by PeterYusuke** [comment]
> This also needs to be rebased since we just merged a PR that updated the dependencies.

Ok, I will.

**Comment by PeterYusuke** [comment]
Fix reviews.

**Review by picklelo — APPROVED** [review]
awesome!

**Pull request merged**